src/System.Management.Automation/engine/Interop/Windows/OpenProcessToken.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Windows
+    {
+        internal const int TOKEN_QUERY = 0x0008;
+
+        [LibraryImport("api-ms-win-core-processsecurity-l1-1-0.dll",  SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static partial bool OpenProcessToken(nint ProcessHandle, int DesiredAccess, out nint TokenHandle);
```

Comment:
Is there a reason why we don't use SafeHandles for HANDLE values? It can ensure that things are always disposed even if the caller explicitly misses the disposal code. It also simplifies the invalid checks through `handle.IsInvalid`.

Personally I'm a fan of using a SafeHandle type because it forces you to wrap the HANDLES properly and can utilise `using ...` to wrap them. The only time I use nint is usually for blittable structures.

Reply:
If SafeHandle had been here before, I would have saved it. This is just the case when we open and close in one place and there is no need to do any allocations more.